vividus-plugin-excel/src/main/java/org/vividus/bdd/transformer/ExcelTableTransformer.java

```diff
@@ -56,33 +60,38 @@ public String transform(String tableAsString, TableParsers tableParsers, TablePr
                 throw new IllegalArgumentException("Sheet with name '" + sheetName + "' does not exist");
             }
             IExcelSheetParser excelSheetParser = new ExcelSheetParser(sheet.get());
-            String lineBreakReplacementPropertyValue = properties.getProperties().getProperty("lineBreakReplacement");
-            String lineBreakReplacement = lineBreakReplacementPropertyValue == null ? ""
-                : lineBreakReplacementPropertyValue;
-            result = extractData(excelSheetParser, properties)
-                    .stream()
-                    .map(e -> e.replace("\n", lineBreakReplacement))
-                    .collect(Collectors.toList());
+            String column = properties.getProperties().getProperty(COLUMN);
+            if (column != null)
+            {
+                notBlank(column, "Table property 'column' is blank");
+                String joinValues = properties.getProperties().getProperty("joinValues");
+                List<String> result = extractData(excelSheetParser, properties);
+                List<String> data = Boolean.parseBoolean(joinValues) ? List.of(String.join(" ", result)) : result;
+                return build(List.of(column), List.of(data), properties);
+            }
+            else
+            {
+                String range = ExtendedTableTransformer.getMandatoryNonBlankProperty(properties, RANGE);
+                Map<String, List<String>> exactDataTable = excelSheetParser.exactDataTable(range);
+                return build(List.copyOf(exactDataTable.keySet()), List.copyOf(exactDataTable.values()), properties);
```

Comment:
why do you need copies?

Reply:
change to  List<String> headers = exactDataTable.keySet().stream().collect(Collectors.toList());
                  List<List<String>> columns = exactDataTable.values().stream().collect(Collectors.toList());